### DISCUSSION TITLE
Sharing code options with .NET MAUI

### DISCUSSION THREAD

**Discussion by juansturla** [dotnet/maui#4447] [Q&A]
A `.NET MAUI `project targets Net6 and the platform variants like net6.0-ios, net6.0-android, and so on.

But what happens when you want to share code or assets between different projects that are not `.NET MAUI`?

Can we still use `Shared Projects` on VS2022 to share any asset between `.NET MAUI` project and any other project? In that case everything would be compiled the shared project using the same target framework that the build project.

What about `Class libraries` that target .NET Standard ?  Which projects can use a .NET Standard library?

What about `Class libraries` that target .NET 5 or 6 ? Which projects can use a .NET 5/6 library?

What about `MAUI Class libraries` ? I guess that is just a `Class libraries` that targets net6.0, net6.0-android [...]. That would mean that projects that can use net6 libraries can use them? 

I want to have a clear idea of what are the alternatives to share Assets between projects

**Comment by Bowman74** [comment]
MAUI projects can still reference, create a use libraries that target .Net Standard. This seems to be the normal way of creating redistributable logic that isn't MAUI specific that we can put into something like a private or public Nuget server. 

Shared projects still exist in the latest VS preview. This is a way to share code within a solution but isn't really the best way to create redistributable logic across solutions and applications into something like Nuget. It is just a way to have multiple projects compile in the same code.

Class libraries that reference platform specific .Net (usually Windows) are not recommended to be used in MAUI, just like with earlier versions of Xamarin. They *May* work, but definitely not recommended.

MAUI Class libraries are intended to create redistributable packages that reference per platform APIs. In particular Android, iOS, Windows and MacOS. If you want to create a redistributable to be used in other MAUI projects that hit platform specific code, this is the way to go. If you want to create redistributable that contains no platform specific code that is intended to be used anywhere, then .Net Standard or similar should be considered. 

For new projects .Net 5 or 6 are also possible options for cross platform redistributables. They are generally recommended over .Net Standard 2.1. This page has a good discussion on when to use one over the other, though it seems like in some case either is a viable solution.

https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-1
